account_invoice_discount_date/models/account_move.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 Moduon Team S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    discount_date = fields.Date(
+        compute="_compute_discount_date",
+        inverse="_inverse_discount_date",
+        store=True,
+        help="Last date at which the discounted amount must be paid in order "
+        "for the Early Payment Discount to be granted",
+    )
+
+    @api.depends("line_ids.discount_date")
+    def _compute_discount_date(self):
+        for record in self:
+            d_dates = sorted(
+                record.line_ids.filtered("discount_date").mapped("discount_date")
+            )
+            record.discount_date = d_dates and sorted(d_dates)[0] or None
+
+    def _inverse_discount_date(self):
+        """When set discount_date set on all move lines with date maturity"""
+        for record in self:
+            for line in record.line_ids:
+                line.discount_date = line.date_maturity and record.discount_date
```

Comment:
If this method were right, it'd have a bug:

```suggestion
                line.discount_date = line.date_maturity and line.discount_date
```

However, `_compute_discount_date()` depends on the value of `line_ids.discount_date`, so the normal thing would be that an inverse method on `discount_date` would write in those fields instead.

But... an inverse field writing on itself? It's the 1st time I see this. 🤔  I think we could approach this better.

It seems to me that what you're trying to achieve is to forbid users from setting a `discount_date` if `date_maturity` is empty, right? If so, the standard way to do that would be:
1. Make `_compute_discount_date` depend on `date_maturity`, and set `discount_date = False` if `date_maturity` is `False`.
2. In the view, make `discount_date` readonly or invisible when `date_maturity == False`.

Reply:
I'll explan what I want to do:

1. If you set `discount_date` on the invoice, the **lines** with `date_maturity` will write its `discount_date` with the `discount_date` you put on the **invoice**. (This is where `_inverse_discount_date` triggers)
2. If you change the `discount_date` of a **line** that has `date_maturity` (found a bug here), get the lower `discount_date` of its **lines** and if it's different than the `discount_date` of the invoice, update it (and then trigger `_inverse_discount_date` again)

Regarding the bug if found, this works as I expected.

Reply:
Sorry I completely misunderstood the code! It's OK indeed.